client/src/main/java/io/pravega/client/netty/impl/RawClient.java

```diff
@@ -56,12 +56,24 @@ public void process(Reply reply) {
                 if (hello.getLowVersion() > WireCommands.WIRE_VERSION || hello.getHighVersion() < WireCommands.OLDEST_COMPATIBLE_VERSION) {
                     closeConnection(new IllegalStateException("Incompatible wire protocol versions " + hello));
                 }
+            } else if (reply instanceof WireCommands.WrongHost) {
+                log.info("WrongHost");
+                closeConnection(new ConnectionFailedException(reply.toString()));
+            } else if (reply instanceof WireCommands.TableSegmentNotEmpty) {
+                log.info("TableSegmentNotEmpty");
+                closeConnection(new ConnectionFailedException(reply.toString()));
             } else {
                 log.debug("Received reply {}", reply);
                 reply(reply);
             }
         }
 
+        @Override
+        public void tableSegmentNotEmpty(WireCommands.TableSegmentNotEmpty tableSegmentNotEmpty) {
```

Comment:
This method is never going to be invoked. `process` above is handling the reply and it does not call this method.